### PR TITLE
fix: change booking page filter ui to match Figma

### LINKED ISSUE #11586: [CAL-2541] Fix the UX/UI of our filters on the booking page
**Current problem**

The current filters jump between left and right aligned on tablet vs desktop and don't align with design.

**Solution**

Match the filter UI/UX to Figma.

[View in Figma](https://www.figma.com/file/UajFu4M1APhn2AywAEJkJr/New-Features?type=design&node-id=7869%3A93929&mode=dev)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2541](https://linear.app/calcom/issue/CAL-2541/fix-the-uxui-of-our-filters-on-the-booking-page)</sub>

### PR DESCRIPTION
**Pull request by asdotdev**
## What does this PR do?

Fixes #11586

[Loom Link](https://www.loom.com/share/f9dd91f18dba4a15ab0fbb64dd314cf8?sid=4e4ec04e-c2b0-4509-acce-23255442da1b)

## Requirement/Documentation

[Figma Link](https://www.figma.com/file/UajFu4M1APhn2AywAEJkJr/New-Features?type=design&node-id=7869%3A93929&mode=dev)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Open the booking page and match the UI with Figma.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent-sized PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

## Note

I have added "filters" in common.json for only en
```diff
+ "filters": "Filters",
```

### CODE REVIEW

File: apps/web/public/static/locales/en/common.json [hunk 1]
```diff
@@ -1843,6 +1843,7 @@
   "review_event_type": "Review Event Type",
   "looking_for_more_analytics": "Looking for more analytics?",
   "looking_for_more_insights": "Looking for more Insights?",
+  "filters": "Filters",
```
Comment (amitabh-sahu) [inline apps/web/public/static/locales/en/common.json:1846]: I can also add this field in some other language's common.js files if needed so let me know.

File: packages/features/bookings/components/FiltersContainer.tsx
Comment (garg3133) [inline packages/features/bookings/components/FiltersContainer.tsx:1]: This file contains two components. Maybe spread them out into two separate files?

File: apps/web/pages/bookings/[status].tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { z } from "zod";
 
 import { WipeMyCalActionButton } from "@calcom/app-store/wipemycalother/components";
 import { getLayout } from "@calcom/features/MainLayout";
-import { FiltersContainer } from "@calcom/features/bookings/components/FiltersContainer";
+import { FiltersContainer, FilterToggleCta } from "@calcom/features/bookings/components/FiltersContainer";
```
Comment (garg3133) [inline apps/web/pages/bookings/[status].tsx:9]: I think just `FiltersToggle` would be a better name for the new component. What do you think?

File: apps/web/public/static/locales/en/common.json (hunk 1 above)
Comment (Udit-takkar) [reply apps/web/public/static/locales/en/common.json:1846]: No it's fine. you only have to add this key value in /en/common.json. we'll add to other languages later

File: packages/features/bookings/components/FiltersContainer.tsx [hunk 3]
```diff
@@ -92,51 +82,40 @@ const PeopleFilter = () => {
...
-                  }}>
-                  {t(option.label)}
-                </DropdownItem>
-              </DropdownMenuItem>
-            ))}
-          </DropdownMenuContent>
-        </Dropdown>
+    <button
+      onClick={toggleFiltersVisibility}
+      className="hover:border-emphasis border-default text-default hover:text-emphasis mb-4 flex h-9 max-h-72 items-center justify-between whitespace-nowrap rounded-md border px-3 py-2 text-sm hover:cursor-pointer focus:border-neutral-300 focus:outline-none focus:ring-2 focus:ring-neutral-800 focus:ring-offset-1">
```
Comment (Udit-takkar) [inline packages/features/bookings/components/FiltersContainer.tsx:109]: We already have Button component in @calcom/ui. we can use   Button
with color="secondary" as prop

File: packages/features/bookings/BookingMultiFiltersStore.ts [hunk 4]
```diff
@@ -1,52 +1,15 @@
...
-    } else {
-      set({ addFilterOptions: [...availableOptions, option] });
-    }
-  },
+  isFiltersVisible: false,
+  toggleFiltersVisibility: () =>
+    set((state) => ({
+      ...state,
+      isFiltersVisible: !get().isFiltersVisible,
+    })),
```
Comment (Udit-takkar) [inline packages/features/bookings/BookingMultiFiltersStore.ts:14]: the new implementation doesn't require zustand store. you have only created one method for toggling visibility. we can probably remove this and use useState instead

File: packages/features/bookings/components/FiltersContainer.tsx [hunk 5]
```diff
@@ -92,51 +82,40 @@ const PeopleFilter = () => {
...
-    shallow,
-  ]);
+  return isFiltersVisible ? (
+    <div className="no-scrollbar flex w-full space-x-2 overflow-x-scroll rtl:space-x-reverse">
+      <PeopleFilter />
+      <EventTypeFilter />
+      <TeamsFilter />
+    </div>
+  ) : null;
+}
```
Comment (Udit-takkar) [inline packages/features/bookings/components/FiltersContainer.tsx:94]: NIT: we can use auto animate for animation here whenever filters are visible.

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
few more things

There is no "All Event Types" in event type filter or "All users" in people filter 

The colors don't match from figma. "People: All" don't have the same color. "People" should be lighter

Screenshot 2023-11-26 at 2 17 31 PM

File: apps/web/public/static/locales/en/common.json (hunk 1 above)
Comment (amitabh-sahu) [reply apps/web/public/static/locales/en/common.json:1868]: Ok

I have added another field:
```dif
"all_event_types_filter_label": "All Event Types",
```

File: packages/features/bookings/components/FiltersContainer.tsx
Comment (amitabh-sahu) [reply packages/features/bookings/components/FiltersContainer.tsx:1]: I have separated the components into two separate files.

File: apps/web/pages/bookings/[status].tsx (hunk 2 above)
Comment (amitabh-sahu) [reply apps/web/pages/bookings/[status].tsx]: I have changed the name of the component.

File: packages/features/bookings/components/FiltersContainer.tsx (hunk 3 above)
Comment (amitabh-sahu) [reply packages/features/bookings/components/FiltersContainer.tsx]: I have made this change.

File: packages/features/bookings/BookingMultiFiltersStore.ts (hunk 4 above)
Comment (amitabh-sahu) [reply packages/features/bookings/BookingMultiFiltersStore.ts]: I have removed this file and used useState

File: packages/features/bookings/components/FiltersContainer.tsx (hunk 5 above)
Comment (amitabh-sahu) [reply packages/features/bookings/components/FiltersContainer.tsx]: Used useAutoAnimate

**Comment by amitabh-sahu** [comment]
> few more things
> 
> There is no "All Event Types" in event type filter or "All users" in people filter
> 
> The colors don't match from figma. "People: All" don't have the same color. "People" should be lighter

I have made these changes and matched them from Figma.

I also separated "PeopleFilter" component into a separate file

**Comment by Udit-takkar** [comment]
> I have made these changes and matched them from Figma.
> 
> I also separated "PeopleFilter" component into a separate file

Awesome. Great work

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**